src/lib.rs

```diff
@@ -37,6 +37,7 @@
 //! the client helper as `Client` in the protocol module you're working with (e.g.,
 //! [`pipeline::Client`]), and the server helper as `Server` in the same place.
 #![deny(missing_docs)]
+#![feature(type_alias_impl_trait)]
```

Comment:
I wonder if you can feature flag this, a bit unfortunate since this doesn't seem like its going to hit stable anytime soon :(

Reply:
I _really_ want this PR to land, but yeah, not looking to stabilize for a while. It would essentially be an unstable feature flag, which isn't great.. :/

Reply:
I wonder if you could do this with some sort of generic or something maybe but yeah, go head we can fix later but we should probably not get things stuck on the nightly drug. I just asked in zulip about this feature not gonna be anytime soon :(

Reply:
So you're proposing we fold this under, like `feature = "nightly-nobox"`? I don't think that'll work, because then the feature wouldn't be additive. If something that depends on `tokio-tower` without the feature somehow got compiled _with_ the feature, it might not compile.

Reply:
yeah im not sure but happy to not block this now for it.

Reply:
I think putting a version out there with non-additive features is just asking for downstream users to have issues sadly. I think it's fine to keep the `Box` version for now, and then have an explicit `git` opt-in for those (like me) who want to squeeze the last little bit out.